libraries/ESP8266WiFi/src/WiFiClient.cpp

```diff
@@ -137,6 +137,22 @@ int WiFiClient::connect(const String& host, uint16_t port)
     return connect(host.c_str(), port);
 }
 
+int WiFiClient::connectAsync(const char* host, uint16_t port, bool* waiting) // Non blocking
+{
+    if (waiting != nullptr)
+        *waiting = false;
+    if (connected())
```

Comment:
If a Client is connected(), another `connect` call will disconnect it and open a new one.  This doesn't.